src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkTask.java

```diff
@@ -71,9 +71,11 @@ public void put(Collection<SinkRecord> records) {
       log.warn(
           "Write of {} records failed, remainingRetries={}",
           records.size(),
-          remainingRetries,
-          sqle
+          remainingRetries
       );
+      for (Throwable e: sqle) {
+        log.warn(e.getMessage());
+      }
```

Comment:
Ok, even though I suggested this approach, it now occurs to me that the separate warning messages may be difficult to reconcile if there are other concurrent log messages from other tasks, connectors, or the worker. What do you think about building a string of these messages delimited by the system line delimiter, and doing this before line 71 and include it in the one log message, something like:

```
Write of {} records failed, remaining retries={}, with {} reasons: {}
```

Thoughts?

Reply:
Do we have any concern on the size of this error message string? On another hand, this log entry is already separated from other log entry in this task, so we might already have this issue anyway.

What if we generate a task_id and bind it to the logger? So that all messages for this task has same task id, users can just filter by task_id.

Reply:
It's possible. I don't know what cardinalities are typical. An alternative is to keep them separate log messages but to clearly denote on each log message (including the first one on line 71) info about the number of messages that are all related. So, if there are `n` exceptions, something like:

```
Write of {} records failed, remaining retries={}, with {} reason 1/{}: {}
```
and add `n` and `sqle.getMessage()`, before `sqle` in the log message parameters. The other log messages might be:

```
Write record failure reason {}/{}: {}
```
with the appropriate parameters (e.g., message #, total message count, error message).

Reply:
I tried the approach concatenating all messages together and log them all at once. It works for me. I'm sure memory usage increased a little bit, but it's not causing any obvious problem or OOM.

I'll make change accordingly.